subiquity/client/controllers/ubuntu_pro.py

```diff
@@ -70,6 +70,7 @@ def check_token(self, token: str,
         async def inner() -> None:
             answer = await self.endpoint.check_token.GET(token)
             if answer.status == TokenStatus.VALID_TOKEN:
+                await self.endpoint.POST(UbuntuProInfo(token=token))
```

Comment:
Doesn't this mean that if the user enables pro and then goes back and selects "do this later" they will still have pro enabled? That doesn't seem quite right.

Reply:
No, because it would go through this code again:

```python
    def upgrade_yes_no_done(self, form: UpgradeYesNoForm) -> None:
        """ If skip is selected, move on to the next screen.
        Otherwise, show the form requesting the contract token. """
        if form.skip.value:
            self.controller.done("")
```
which in turn would do:
```python
 await self.endpoint.POST(UbuntuProInfo(token=""))
```

but if this isn't clear, I can try to refactor again.